src/library/autodj/autodjprocessor.cpp

```diff
@@ -817,61 +998,196 @@ void AutoDJProcessor::calculateTransition(DeckAttributes* pFromDeck,
     double fromTrackDuration = pFromDeck->duration();
     double toTrackDuration = pToDeck->duration();
 
-    double fromTrackOutroStart = getOutroStartPosition(pFromDeck);
-    if (fromTrackOutroStart <= 0.0) {
-        fromTrackOutroStart = fromTrackDuration;
+    VERIFY_OR_DEBUG_ASSERT(fromTrackDuration > 0) {
+        // Playing Track has no duration. This should not happen, because short
+        // tracks are skipped after load. Play ToDeck immediately.
+        pFromDeck->fadeBeginPos = 0;
+        pFromDeck->fadeEndPos = 0;
+        pToDeck->startPos = kKeepPosition;
+        return;
     }
-
-    double fromTrackOutroEnd = getOutroEndPosition(pFromDeck);
-    if (fromTrackOutroEnd <= 0.0) {
-        fromTrackOutroEnd = fromTrackDuration;
+    VERIFY_OR_DEBUG_ASSERT(toTrackDuration > 0) {
```

Comment:
So what do you propose? Just returning here without disabling AutoDJ? Or replacing the VERIFY_AND_DEBUG_ASSERT with a simple `if`? Or loading the next track in the playlist?

Reply:
I think we need too load a new track and not stop AutoDj. 
I just wonder why this has not already happened. Mabye because of two consecutive wma tracks. Or because it was the first track? 
We need to investigate.

Reply:
I am in doubt this shouldn't be an assertion. Why would a track duration ever be <= 0 unless it is a corrupt file?

Reply:
It violates the programming pattern "assert the impossible". You can compile Mixxx that it stops on violated assertions. This should never happen. So we can either make it impossible or remove the assertion.

Reply:
I think I have fixed this with f13bb5d, however I have no tracks that triggered this assertion before so I cannot test it.